### PR TITLE
platform: restructure visualc module

### PR DESCRIPTION
**Pull request by compnerd**
Restructure the `visualc` module into `vcruntime` in order to help
expose the various components (SAL, vcruntime, ucrt, corecrt, STL) for C++
modularization.  Include the `stdint.h` textually to deal with
redefinition of types in clang resources and MSVC.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-installer-scripts#135

@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-installer-scripts/pull/135

@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test Linux platfrom

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

File: test/Interop/Cxx/class/Inputs/memory-layout.h [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #ifndef TEST_INTEROP_CXX_CLASS_INPUTS_MEMORY_LAYOUT_H
 #define TEST_INTEROP_CXX_CLASS_INPUTS_MEMORY_LAYOUT_H
 
-#include <stddef.h>
-#include <stdint.h>
+#include <cstddef>
+#include <cstdint>
```
Comment (buttaface) [inline test/Interop/Cxx/class/Inputs/memory-layout.h:5]: @compnerd, it appears this change [broke the C++ Interop test `memory-layout-silgen` on the Android CI](https://ci-external.swift.org/job/oss-swift-RA-linux-ubuntu-16.04-android-arm64/lastFailedBuild/console). I confirmed by running this test with the last Jun. 26 trunk snapshot without this patch, which passed, then after applying this pull, which made the test fail:
```
<module-includes>:1:10: note: in file included from <module-includes>:1:
#include "memory-layout.h"
         ^
/home/ubuntu/jenkins/workspace/oss-swift-RA-linux-ubuntu-16.04-android-arm64/swift/test/Interop/Cxx/class/Inputs/memory-layout.h:14:8: error: missing '#include <sys/types.h>'; 'size_t' must be declared before it is used
inline size_t sizeOfPrivateMemberLayout() {
       ^
/home/ubuntu/jenkins/workspace/oss-swift-RA-linux-ubuntu-16.04-android-arm64/buildbot_linux/swift-linux-x86_64/lib/swift/clang/include/stddef.h:46:23: note: declaration here is not visible
typedef __SIZE_TYPE__ size_t;
                      ^
```
I really don't want to look into these interoperability issues, so if you or @drodriguez know what's going on here, let's get this fixed, or I can just disable it for Android.

Comment (compnerd) [reply test/Interop/Cxx/class/Inputs/memory-layout.h:5]: Hmm, it seems like the module.modulemap for android might not be correct.  It is attributing `size_t` to `sys/types.h` as that is likely the header that is importing `stddef.h` first.  An explicit module should be added for `C.stddef` which should home `size_t` properly.

**Pull request merged**